<?xml version="1.0" encoding="utf-8"?><testsuites><testsuite name="pytest" errors="0" failures="4" skipped="0" tests="8" time="0.324" timestamp="2023-12-08T00:08:40.971850" hostname="DESKTOP-642O1ET"><testcase classname="test_library_management" name="test_successful_checkout" time="0.002" /><testcase classname="test_library_management" name="test_checkout_unavailable_book" time="0.001" /><testcase classname="test_library_management" name="test_max_books_per_checkout" time="0.001" /><testcase classname="test_library_management" name="test_return_no_late_fee" time="0.002"><failure message="assert 15 == 0">def test_return_no_late_fee():
        library = Library()
        library.checked_out_books = [{'index': 1,'title': '1984', 'quantity': 1, 'due_date': datetime.now() - timedelta(days=15), 'late_fees': 0}]
        total_late_fees = return_books_checkout(0,library,1,0)
&gt;       assert total_late_fees == 0
E       assert 15 == 0

test_library_management.py:36: AssertionError</failure></testcase><testcase classname="test_library_management" name="test_return_late_fee" time="0.001"><failure message="assert 11 == 12&#10; +  where 11 = &lt;main.Book object at 0x000001B1F1742B10&gt;.quantity">def test_return_late_fee():
        library = Library()
        library.checked_out_books = [{'index':1, 'title': '1984', 'quantity': 1, 'due_date': datetime.now() - timedelta(days=15), 'late_fees': 0}]
        total_late_fees = return_books_checkout(0,library,1,0)
        assert total_late_fees == 15  # Se espera un cargo por retraso de $1
&gt;       assert library.books[0].quantity == 12  # Verifica que se increment√≥ el inventario
E       assert 11 == 12
E        +  where 11 = &lt;main.Book object at 0x000001B1F1742B10&gt;.quantity

test_library_management.py:44: AssertionError</failure></testcase><testcase classname="test_library_management" name="test_invalid_book_index_checkout" time="0.001"><failure message="IndexError: list index out of range">def test_invalid_book_index_checkout():
        library = Library()
        selections = [{'book_index': len(library.books), 'quantity': 1}]
&gt;       total_late_fees = library.checkout_books(selections)

test_library_management.py:50: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = &lt;main.Library object at 0x000001B1F1732250&gt;, selections = [{'book_index': 2, 'quantity': 1}]

    def checkout_books(self, selections):
        total_late_fees = 0
        count = 1
        self.checked_out_books_tmp = []
        for selection in selections:
    
&gt;           book = self.books[selection['book_index']]
E           IndexError: list index out of range

main.py:33: IndexError</failure></testcase><testcase classname="test_library_management" name="test_invalid_quantity_return" time="0.001" /><testcase classname="test_library_management" name="test_accumulated_late_fees" time="0.002"><failure message="assert 60 == 4">def test_accumulated_late_fees():
        library = Library()
        library.checked_out_books = [{'index':0,'title': '1984', 'quantity': 2, 'due_date': datetime.now() - timedelta(days=15), 'late_fees': 0},{'index':1,'title': '1984', 'quantity': 2, 'due_date': datetime.now() - timedelta(days=15), 'late_fees': 0}]
        total_late_fees = return_books_checkout(0,library,2,0) + return_books_checkout(1,library,2,0)
&gt;       assert total_late_fees == 4
E       assert 60 == 4

test_library_management.py:67: AssertionError</failure></testcase></testsuite></testsuites>